RecoCTPPS/TotemRPLocal/interface/TotemRPLocalTrackFitterAlgorithm.h

```diff
@@ -0,0 +1,73 @@
+/****************************************************************************
+*
+* This is a part of TOTEM offline software.
+* Authors:
+* 	Hubert Niewiadomski
+*   Jan Kašpar (jan.kaspar@gmail.com)
+*
+****************************************************************************/
+
+#ifndef RecoCTPPS_TotemRPLocal_TotemRPLocalTrackFitterAlgorithm
+#define RecoCTPPS_TotemRPLocal_TotemRPLocalTrackFitterAlgorithm
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/CTPPSReco/interface/TotemRPRecHit.h"
+#include "DataFormats/CTPPSReco/interface/TotemRPLocalTrack.h"
+
+#include "Geometry/VeryForwardGeometryBuilder/interface/TotemRPGeometry.h"
+#include "Geometry/VeryForwardRPTopology/interface/RPTopology.h"
+
+#include "TVector3.h"
+#include "TVector2.h"
+
+#include <unordered_map>
+
+//----------------------------------------------------------------------------------------------------
+
+/**
+ *\brief Algorithm for fitting tracks through a single RP.
+ **/
+class TotemRPLocalTrackFitterAlgorithm
+{
+  public:
+    TotemRPLocalTrackFitterAlgorithm(const edm::ParameterSet &conf);
+
+    /// performs the track fit, returns true if successful
+    bool FitTrack(const edm::DetSetVector<TotemRPRecHit> &hits, double z_0, const TotemRPGeometry &tot_geom, TotemRPLocalTrack &fitted_track);
+
+    /// Resets the reconstruction-data cache.
+    void Reset();
```

Comment:
Why not pass the geometry into the routine as an argument so that you can update the cache here?